### ISSUE TITLE
Extending for (already analyzed) Lichess games? (enhancement)

### ISSUE THREAD

**Issue by rwbc** [fsmosca/Python-Chess-Scripts#8]
I think with Lichess as the biggest online chess server (and open source too!) it would be nice to support its format too.
You probably know Lichess has a auto-analyze feature for checking played games later with a fast SF search and comment all games (if you have an account).
This is saved to thegames and available for download either by the Lichess api via filters or via the Lichess game database.

https://lichess.org/api#operation/apiGamesUser
https://database.lichess.org/

A few people even have set up an UI for making it easier to send requests for games download via the api commands.

https://lichessgamedownload.netlify.app/
https://fbserv.herokuapp.com/games.html?

Of course creating graphs only makes sense for the already analyzed games, but you can filter them already before downloading. (not with the total databse per month download)

Here is an example of its format:

```
[Event "RT-Schnellschachturnier 31"]
[Site "https://lichess.org/HPpt04kq"]
[Date "2020.12.16"]
[White "Bruto130199"]
[Black "hot_sun"]
[Result "0-1"]
[UTCDate "2020.12.16"]
[UTCTime "20:49:15"]
[WhiteElo "2140"]
[BlackElo "2425"]
[WhiteRatingDiff "-3"]
[BlackRatingDiff "+3"]
[Variant "Standard"]
[TimeControl "900+5"]
[ECO "B23"]
[Termination "Normal"]

1. e4 { [%eval 0.0] [%clk 0:15:00] } 1... c5 { [%eval 0.0] [%clk 0:15:00] } 2. Nc3 { [%eval 0.12] [%clk 0:14:58] } 2... a6 { [%eval 0.29] [%clk 0:15:03] } 3. a4 { [%eval -0.17] [%clk 0:15:02] } 3... e6 { [%eval 0.13] [%clk 0:15:06] } 4. Bc4 { [%eval -0.31] [%clk 0:15:06] } 4... Nc6 { [%eval -0.33] [%clk 0:15:07] } 5. d3 { [%eval -0.44] [%clk 0:15:07] } 5... Nf6 { [%eval -0.22] [%clk 0:15:08] } 6. f4 { [%eval -0.73] [%clk 0:15:03] } 6... d5 { [%eval -0.82] [%clk 0:15:11] } 7. Ba2 { [%eval -0.6] [%clk 0:15:06] } 7... dxe4 { [%eval -0.36] [%clk 0:14:34] } 8. Nxe4 { [%eval -0.64] [%clk 0:15:08] } 8... Nb4 { [%eval -0.16] [%clk 0:13:17] } 9. Bb3 { [%eval -0.28] [%clk 0:15:08] } 9... Be7 { [%eval -0.31] [%clk 0:12:34] } 10. Nf3 { [%eval -0.2] [%clk 0:15:10] } 10... Qc7 { [%eval 0.0] [%clk 0:12:38] } 11. O-O { [%eval -0.89] [%clk 0:15:13] } 11... Nxe4 { [%eval -0.74] [%clk 0:12:42] } 12. dxe4 { [%eval -0.85] [%clk 0:15:17] } 12... c4 { [%eval -1.08] [%clk 0:12:46] } 13. Ba2 { [%eval -1.0] [%clk 0:15:21] } 13... O-O { [%eval -0.92] [%clk 0:12:28] } 14. c3 { [%eval -1.58] [%clk 0:15:24] } 14... Nxa2 { [%eval -1.25] [%clk 0:12:31] } 15. Rxa2 { [%eval -1.24] [%clk 0:15:29] } 15... b6 { [%eval -1.23] [%clk 0:11:49] } 16. Be3 { [%eval -1.16] [%clk 0:15:30] } 16... Bb7 { [%eval -1.18] [%clk 0:11:48] } 17. e5 { [%eval -1.21] [%clk 0:15:09] } 17... Rad8 { [%eval -1.0] [%clk 0:11:28] } 18. Qe2 { [%eval -1.49] [%clk 0:15:10] } 18... Rd3 { [%eval -1.47] [%clk 0:11:04] } 19. a5 { [%eval -2.29] [%clk 0:14:57] } 19... bxa5 { [%eval -2.17] [%clk 0:10:23] } 20. Nd4 { [%eval -2.9] [%clk 0:14:55] } 20... Bc5 { [%eval -2.76] [%clk 0:10:07] } 21. Rfa1 { [%eval -3.42] [%clk 0:14:49] } 21... Bb6 { [%eval -3.31] [%clk 0:10:09] } 22. Rd1 { [%eval -4.99] [%clk 0:14:25] } 22... Bd5 { [%eval -2.82] [%clk 0:09:43] } 23. Raa1 { [%eval -2.74] [%clk 0:14:22] } 23... Rxd1+ { [%eval -1.87] [%clk 0:09:45] } 24. Qxd1 { [%eval -2.67] [%clk 0:14:18] } 24... f6 { [%eval -2.11] [%clk 0:09:21] } 25. exf6 { [%eval -2.49] [%clk 0:14:13] } 25... Rxf6 { [%eval -2.17] [%clk 0:09:26] } 26. Qd2 { [%eval -1.82] [%clk 0:14:10] } 26... h5 { [%eval -1.61] [%clk 0:09:19] } 27. Rf1 { [%eval -1.39] [%clk 0:14:03] } 27... h4 { [%eval -1.48] [%clk 0:09:21] } 28. h3 { [%eval -1.47] [%clk 0:14:07] } 28... Rg6 { [%eval -0.26] [%clk 0:08:50] } 29. Nf3 { [%eval -9.47] [%clk 0:14:08] } 29... Qxf4 { [%eval -9.64] [%clk 0:08:13] } 30. Bxb6 { [%eval -10.09] [%clk 0:12:28] } 30... Qxd2 { [%eval -9.9] [%clk 0:07:42] } 31. Nxd2 { [%eval -10.42] [%clk 0:12:30] } 31... Rxg2+ { [%eval -10.33] [%clk 0:07:47] } 32. Kh1 { [%eval -10.39] [%clk 0:12:34] } 32... Rxd2+ { [%eval -10.48] [%clk 0:07:51] } 33. Kg1 { [%eval -10.42] [%clk 0:12:37] } 33... Rg2+ { [%eval -10.56] [%clk 0:07:56] } 34. Kh1 { [%eval -10.62] [%clk 0:12:41] } 34... Rxb2+ { [%eval -10.35] [%clk 0:08:01] } 35. Kg1 { [%eval -10.04] [%clk 0:12:46] } 35... Rxb6 { [%eval -10.48] [%clk 0:08:05] } 0-1
```

A dozen of selected games (only rated rapid for a certain period)
[lichess_hot_sun_2020-12-26.zip](https://github.com/fsmosca/Python-Chess-Scripts/files/5744465/lichess_hot_sun_2020-12-26.zip)

**Comment by fsmosca** [comment]
I will implement this with `--lichess` flag.

**Comment by fsmosca** [comment]
`plot.py --input lichess1.pgn --lichess`

![](https://i.imgur.com/UDkSecw.png)

**Comment by fsmosca** [comment]
Lichess games with mate scores.

v0.18.0

`plot.py --input ./pgn/lichess2.pgn --lichess --min-eval-limit -400 --max-eval-limit 400`

![mate](https://i.imgur.com/C2fMYMM.png)

**Comment by rwbc** [comment]
I get an error with a lichess pgn starting at line 135, for this game (#scores = game3 in the full pgn)
Probably I should have included more examples for lichess games in the beginning.
I have attached one for a closed account of mine with 3-0 games (236 games)

```
[Event "Rated Blitz game"]
[Site "https://lichess.org/0qEzbV0U"]
[Date "2020.05.30"]
[White "andreygubkin"]
[Black "osterei"]
[Result "1-0"]
[UTCDate "2020.05.30"]
[UTCTime "18:52:56"]
[WhiteElo "2131"]
[BlackElo "2094"]
[WhiteRatingDiff "+5"]
[BlackRatingDiff "-5"]
[Variant "Standard"]
[TimeControl "180+0"]
[ECO "A45"]
[Termination "Normal"]

1. d4 { [%eval 0.25] [%clk 0:03:00] } 1... Nf6 { [%eval 0.22] [%clk 0:03:00] } 2. Bf4 { [%eval 0.0] [%clk 0:03:00] } 2... c5 { [%eval 0.18] [%clk 0:02:59] } 3. c3 { [%eval -0.5] [%clk 0:02:58] } 3... Nc6 { [%eval 0.63] [%clk 0:02:58] } 4. e3 { [%eval 0.0] [%clk 0:02:58] } 4... d5 { [%eval 0.0] [%clk 0:02:57] } 5. Bd3 { [%eval -0.21] [%clk 0:02:57] } 5... e6 { [%eval 0.0] [%clk 0:02:56] } 6. Nd2 { [%eval -0.26] [%clk 0:02:56] } 6... Bd6 { [%eval 0.0] [%clk 0:02:55] } 7. Bxd6 { [%eval 0.0] [%clk 0:02:55] } 7... Qxd6 { [%eval 0.0] [%clk 0:02:54] } 8. f4 { [%eval 0.2] [%clk 0:02:55] } 8... c4 { [%eval 0.13] [%clk 0:02:53] } 9. Bc2 { [%eval 0.27] [%clk 0:02:50] } 9... b5 { [%eval 0.33] [%clk 0:02:52] } 10. Ngf3 { [%eval 0.47] [%clk 0:02:49] } 10... Bb7 { [%eval 0.53] [%clk 0:02:50] } 11. O-O { [%eval 0.35] [%clk 0:02:46] } 11... h6 { [%eval 0.77] [%clk 0:02:44] } 12. Ne5 { [%eval 0.69] [%clk 0:02:45] } 12... O-O { [%eval 1.43] [%clk 0:02:37] } 13. Ndf3 { [%eval 0.26] [%clk 0:02:44] } 13... Ne4 { [%eval 0.28] [%clk 0:02:35] } 14. Qb1 { [%eval -0.62] [%clk 0:02:37] } 14... f5 { [%eval 1.16] [%clk 0:02:34] } 15. Nxc6 { [%eval 0.79] [%clk 0:02:33] } 15... Bxc6 { [%eval 0.9] [%clk 0:02:32] } 16. Bxe4 { [%eval 0.91] [%clk 0:02:33] } 16... dxe4 { [%eval 0.92] [%clk 0:02:28] } 17. Ne5 { [%eval 0.9] [%clk 0:02:32] } 17... Be8 { [%eval 1.42] [%clk 0:02:27] } 18. Qd1 { [%eval 1.28] [%clk 0:02:30] } 18... Rb8 { [%eval 1.63] [%clk 0:02:26] } 19. g4 { [%eval 1.06] [%clk 0:02:24] } 19... a5 { [%eval 1.41] [%clk 0:02:19] } 20. g5 { [%eval 0.94] [%clk 0:02:15] } 20... h5 { [%eval 7.21] [%clk 0:02:17] } 21. h4 { [%eval 2.69] [%clk 0:02:13] } 21... g6 { [%eval 2.86] [%clk 0:02:15] } 22. Kg2 { [%eval 2.26] [%clk 0:02:11] } 22... Kg7 { [%eval 2.62] [%clk 0:02:15] } 23. Kg3 { [%eval 1.37] [%clk 0:02:10] } 23... Bf7 { [%eval 2.53] [%clk 0:02:09] } 24. b3 { [%eval 2.38] [%clk 0:02:09] } 24... Rfc8 { [%eval 3.46] [%clk 0:02:07] } 25. bxc4 { [%eval 3.09] [%clk 0:02:07] } 25... bxc4 { [%eval 3.23] [%clk 0:02:06] } 26. Rb1 { [%eval 2.72] [%clk 0:02:06] } 26... Qa3 { [%eval 3.04] [%clk 0:01:55] } 27. Qc2 { [%eval 3.33] [%clk 0:01:59] } 27... Be8 { [%eval 3.26] [%clk 0:01:48] } 28. Rxb8 { [%eval 3.45] [%clk 0:01:53] } 28... Rxb8 { [%eval 3.34] [%clk 0:01:47] } 29. Rb1 { [%eval 0.0] [%clk 0:01:33] } 29... Ba4 { [%eval 0.0] [%clk 0:01:44] } 30. Qc1 { [%eval 0.1] [%clk 0:01:24] } 30... Qxc1 { [%eval 0.0] [%clk 0:01:42] } 31. Rxc1 { [%eval 0.0] [%clk 0:01:23] } 31... Rb2 { [%eval 0.16] [%clk 0:01:30] } 32. Nxc4 { [%eval 0.47] [%clk 0:01:21] } 32... Rxa2 { [%eval 0.9] [%clk 0:01:29] } 33. Nxa5 { [%eval 0.0] [%clk 0:01:19] } 33... Re2 { [%eval 0.0] [%clk 0:01:19] } 34. Nc4 { [%eval 0.0] [%clk 0:01:15] } 34... Bc2 { [%eval 0.0] [%clk 0:01:13] } 35. Ra1 { [%eval 0.0] [%clk 0:01:06] } 35... Bd3 { [%eval 0.01] [%clk 0:01:12] } 36. Ne5 { [%eval 0.03] [%clk 0:01:03] } 36... Rxe3+ { [%eval 0.69] [%clk 0:01:08] } 37. Kf2 { [%eval 0.34] [%clk 0:01:02] } 37... Rh3 { [%eval 2.85] [%clk 0:01:07] } 38. Ra7+ { [%eval 1.08] [%clk 0:00:59] } 38... Kf8 { [%eval 2.33] [%clk 0:01:05] } 39. Nxg6+ { [%eval 1.35] [%clk 0:00:58] } 39... Ke8 { [%eval 3.3] [%clk 0:01:03] } 40. Re7+ { [%eval 2.58] [%clk 0:00:54] } 40... Kd8 { [%eval 3.33] [%clk 0:01:02] } 41. Rxe6 { [%eval 2.84] [%clk 0:00:53] } 41... Rf3+ { [%eval 3.15] [%clk 0:01:01] } 42. Ke1 { [%eval -4.58] [%clk 0:00:51] } 42... Re3+ { [%eval 3.99] [%clk 0:00:57] } 43. Kd2 { [%eval 3.07] [%clk 0:00:49] } 43... Re2+ { [%eval 3.2] [%clk 0:00:56] } 44. Kc1 { [%eval 0.0] [%clk 0:00:48] } 44... Rc2+ { [%eval 4.0] [%clk 0:00:55] } 45. Kd1 { [%eval 3.83] [%clk 0:00:47] } 45... Rxc3 { [%eval 3.65] [%clk 0:00:54] } 46. Rd6+ { [%eval 2.07] [%clk 0:00:40] } 46... Kc7 { [%eval 2.02] [%clk 0:00:53] } 47. Rd5 { [%eval 1.94] [%clk 0:00:39] } 47... e3 { [%eval 8.07] [%clk 0:00:51] } 48. Ne5 { [%eval 0.0] [%clk 0:00:31] } 48... Be4 { [%eval 9.25] [%clk 0:00:49] } 49. Rd7+ { [%eval 3.35] [%clk 0:00:21] } 49... Kc8 { [%eval 2.96] [%clk 0:00:47] } 50. Rh7 { [%eval 0.19] [%clk 0:00:21] } 50... Bf3+ { [%eval 18.93] [%clk 0:00:39] } 51. Nxf3 { [%eval 16.98] [%clk 0:00:16] } 51... e2+ { [%eval #17] [%clk 0:00:37] } 52. Kxe2 { [%eval #16] [%clk 0:00:15] } 52... Rc2+ { [%eval #16] [%clk 0:00:36] } 53. Kd3 { [%eval #15] [%clk 0:00:13] } 53... Rf2 { [%eval #15] [%clk 0:00:35] } 54. Ke3 { [%eval #14] [%clk 0:00:11] } 54... Rg2 { [%eval #14] [%clk 0:00:33] } 55. Ne5 { [%eval #13] [%clk 0:00:10] } 55... Rg3+ { [%eval #16] [%clk 0:00:33] } 56. Kf2 { [%eval #17] [%clk 0:00:09] } 56... Rg4 { [%eval #6] [%clk 0:00:32] } 57. Nxg4 { [%eval #5] [%clk 0:00:07] } 57... hxg4 { [%eval #5] [%clk 0:00:31] } 58. Kg3 { [%eval #4] [%clk 0:00:07] } 58... Kd8 { [%eval #4] [%clk 0:00:29] } 59. Rg7 { [%eval #4] [%clk 0:00:06] } 59... Ke8 { [%eval #4] [%clk 0:00:29] } 60. h5 { [%eval #3] [%clk 0:00:06] } 60... Kf8 { [%eval #3] [%clk 0:00:29] } 61. h6 { [%eval #2] [%clk 0:00:06] } 61... Ke8 { [%eval #2] [%clk 0:00:26] } 62. d5 { [%eval #3] [%clk 0:00:06] } 62... Kf8 { [%eval #3] [%clk 0:00:25] } 63. d6 { [%eval #2] [%clk 0:00:06] } 63... Ke8 { [%eval #2] [%clk 0:00:25] } 64. d7+ { [%eval #5] [%clk 0:00:05] } 64... Kd8 { [%eval #5] [%clk 0:00:24] } 65. h7 { [%eval #4] [%clk 0:00:05] } 65... Kc7 { [%eval #4] [%clk 0:00:24] } 66. h8=Q { [%eval #3] [%clk 0:00:04] } 66... Kd6 { [%eval #3] [%clk 0:00:24] } 67. Qh7 { [%eval #4] [%clk 0:00:04] } 67... Kd5 { [%eval #3] [%clk 0:00:23] } 68. Re7 { [%eval #3] [%clk 0:00:03] } 68... Kd4 { [%eval #3] [%clk 0:00:22] } 69. d8=Q+ { [%eval #2] [%clk 0:00:03] } 69... Kc3 { [%eval #2] [%clk 0:00:21] } 70. Rc7+ { [%eval #2] [%clk 0:00:03] } 70... Kb4 { [%eval #2] [%clk 0:00:19] } 71. Qb8+ { [%eval #1] [%clk 0:00:03] } 71... Ka3 { [%eval #1] [%clk 0:00:17] } 72. Ra7# { [%clk 0:00:03] } 1-0
```

```
C:\ChessTools\QualityControl\Plot_PGN>python C:\ChessTools\QualityControl\Plot_PGN\plot.py --lichess --min-eval-limit -800 --max-eval-limit 800 --input C:\ChessTools\QualityControl\Plot_PGN\PGN\lichess_osterei_2020-12-27
.pgn  1>log.txt
Traceback (most recent call last):
  File "C:\ChessTools\QualityControl\Plot_PGN\plot.py", line 436, in <module>
    main()
  File "C:\ChessTools\QualityControl\Plot_PGN\plot.py", line 432, in main
    a.run()
  File "C:\ChessTools\QualityControl\Plot_PGN\plot.py", line 346, in run
    self.plotter(game, output)
  File "C:\ChessTools\QualityControl\Plot_PGN\plot.py", line 238, in plotter
    move_eval = self.get_eval(comment, parent_board.turn, ply, b_eval, w_eval)
  File "C:\ChessTools\QualityControl\Plot_PGN\plot.py", line 135, in get_eval
    split_eval = comment.split('%eval ')[1].split(']')[0]
IndexError: list index out of range
```

[lichess_osterei_2020-12-27.zip](https://github.com/fsmosca/Python-Chess-Scripts/files/5761425/lichess_osterei_2020-12-27.zip)

**Comment by fsmosca** [comment]
Game is over, no eval, I will fix this.

**Comment by fsmosca** [comment]
This is now fixed in v0.22.0.

Tried all the 236 games in lichess_osterei and there are no error messages.
